test/org/zaproxy/zap/extension/ascanrules/HTTPDTestServer.java

```diff
@@ -28,6 +28,12 @@
     
     private List<NanoServerHandler> handlers = 
             new ArrayList<NanoServerHandler>();
+    
+    private NanoServerHandler handler404 = new NanoServerHandler(""){
+		@Override
+		Response serve(IHTTPSession session) {
```

Comment:
Shouldn't this actually set the response code to 404 (something like: https://gist.github.com/tana/3436665#file-myserver-java-L29 .... maybe not a perfect example but you get what I mean). In case the plugin being tested actually cares about the response code.